Feature/pocket feed

bobbyshaw opened the pull request:
Remove manual add item form and pull feed through from pocket.  Gets run upon login and on the hour afterwards.

Also worked on using Readability as a secondary method of getting text content if Goose fails. 

Added title and excerpt from Pocket API response to items table as that's more reliable.

bobbyshaw committed:
Work in progress for a command that pulls in URLs from pocket

- Removed items controller and form field for adding urls manually.

bobbyshaw committed:
Work in progress for a command that pulls in URLs from pocket

- Removed items controller and form field for adding urls manually.

bobbyshaw committed:
Local file id md5 of url not ID.

bobbyshaw committed:
Check for articles to convert to speech and upload to amazon every 15 minutes.

bobbyshaw committed:
Fix useritems relationships.

bobbyshaw committed:
Add useritems relationship to user and correct items relationship.  We can't use the hasManyThrough because it's expecting a user_id column on the items table.

bobbyshaw committed:
Pass useritems to dashboard.

bobbyshaw committed:
Remove debug line.

bobbyshaw committed:
Items template now has user item so we need to get to the item before retrieving values.

bobbyshaw committed:
Move to new way of gettings items from user model.

bobbyshaw committed:
Update podcase feed to:

- Work off username rather than email which we don't have anymore
- As we don't have email, use gravatar image of pocket support instead
- Add excerpt
- Correct S3 urls.

bobbyshaw committed:
Merge branch 'feature/pocket-feed' of github.com:meanbee/articles-to-podcast into feature/pocket-feed

bobbyshaw committed:
This will be slow but synchronise items upon login so that we don't wait an hour for cron.

tgerulaitis commented on app/Services/Pocket.php:
```diff
@@ -0,0 +1,221 @@
+        }
+
+        // Remove previous items which are no longer in the feed
+        $affectedRows = UserItemsModel::whereNotIn('item_id', array_map("md5", $urls))->delete();
```
There's no condition on the user_id here! Does this mean we're removing all items except for the ones that appear in the current user's feed? For each user?

bobbyshaw replied:
Good point.  This was a throw-away line that hasn't been tested atm.

tgerulaitis commented on app/Services/Pocket.php:
```diff
@@ -0,0 +1,221 @@
+                Log::error('Authenticated but lack of permission to access this resource');
+            } elseif($statusCode == 500) {
+                Log::error('Pocket server issue');
+            }
```
You log an error on an unsuccessful response, but still proceed to read json afterwards. Should probably return here?

bobbyshaw replied:
Yup

tgerulaitis commented on app/Services/Pocket.php:
```diff
@@ -0,0 +1,221 @@
+    protected function addItem($url, $title = null, $excerpt = null)
+    {
+        if (ItemModel::find(md5($url))) {
+            return ItemModel::find(md5($url));
```
Will this not make 2 database queries? I'd rewrite this as:

```
if ($item = ItemModel::find(md5($url))) {
    return $item;
}
```

bobbyshaw replied:
Yep

tgerulaitis commented on app/Services/Pocket.php:
```diff
@@ -0,0 +1,221 @@
+     * @return string
+     */
+    protected function getArticleContent($url)
+    {
```
I don't really agree that synchronising Pocket and fetching article text should be the same task. If we get items from Pocket, but fail to fetch their text (because a service is down or something) then we'll have to get items from Pocket again.

bobbyshaw replied:
Yep, this is a migration of the items controller which added a URL and fetched it's content at the same time. So it's MVP atm.

tgerulaitis commented on app/Services/Pocket.php:
```diff
@@ -0,0 +1,221 @@
+          \xF0[\x90-\xBF][\x80-\xBF]{2}      # planes 1-3
+        | [\xF1-\xF3][\x80-\xBF]{3}          # planes 4-15
+        | \xF4[\x80-\x8F][\x80-\xBF]{2}      # plane 16
+        )%xs', '', $string);
```
Literally no idea what's going on here...

bobbyshaw replied:
emoticons are in utf8mb4 which isn't supported in the out of the box mysql so I have to strip any characters that aren't UTF8.

bobbyshaw committed:
Add debugbar.

bobbyshaw committed:
Ignore debugbar logs.

bobbyshaw committed:
Don't synchronise pocket at login, create a separate route and add a link to it in the account area.
Also ensure that we start the login process with no pocket code set from a previous failed login.

bobbyshaw committed:
Order articles by recently updated.

bobbyshaw committed:
Add new failed status for items that can't be fetched.

bobbyshaw committed:
Rework pocket service to:

- Not fetch content upon synchronisation
- Delete users articles when pocket marks them as read
- Delete articles when no longer needed
- Use created and updated at dates from pocket.

bobbyshaw committed:
Add service provider for debugbar.

bobbyshaw committed:
Add new service for extracting content from URLs.

bobbyshaw committed:
Add console command for finding synchronised articles that need content fetching.

bobbyshaw committed:
Add run all command and make this the hourly task.

bobbyshaw committed:
Fix session error.

bobbyshaw committed:
Rename item to article in dashboard.

bobbyshaw committed:
Change links to button in dashboard.

bobbyshaw committed:
Auto-redirect to dashboard from home page if pocket user logged in.

bobbyshaw committed:
Remove duplicate function.

bobbyshaw committed:
Add a bit more info on why we need to convert directly to utf8.

bobbyshaw committed:
Add info on homepage over what the service is and style it centrally.
